src/httpd_jsonapi.c

```diff
@@ -2903,6 +2909,44 @@ jsonapi_reply_library_playlist_tracks(struct httpd_request *hreq)
   return HTTP_OK;
 }
 
+static int
+jsonapi_reply_library_playlist_from_queue(struct httpd_request *hreq)
+{
+  const char *param;
+  int ret = 0;
+  char buf[PATH_MAX+15];
+  char *plsname = NULL;
+
+  if ( (param = evhttp_find_header(hreq->query, "name")) == NULL)
+    {
+      DPRINTF(E_LOG, L_WEB, "Invalid argument, missing 'name'\n");
+      return HTTP_BADREQUEST;
+    }
+
+  if (default_pl_dir == NULL || !allow_modifying_stored_playlists)
+    return HTTP_NOTIMPLEMENTED; 
```

Comment:
Please add a user understandable error log message indicating that the configuration does not allow saving the queue.

And please return a "403" error code (the 403 handling then needs to be added to the function `jsonapi_request`).

Reply:
done.  Annoyingly `libevent2` (at least 2.1.8 on Fedora) doesnt have a define for this so left it in the code as `403`